src/module_ncio.f90

```diff
@@ -79,42 +105,40 @@ module module_ncio
           write_vardata_1d_char, write_vardata_2d_char, write_vardata_3d_char, &
           write_vardata_4d_char, write_vardata_5d_char
   end interface write_vardata
-
+!  ---read netcdf attribute
   interface read_attribute
      module procedure read_attribute_r4_scalar, read_attribute_int_scalar,&
           read_attribute_r8_scalar, read_attribute_r4_1d,&
           read_attribute_int_1d, read_attribute_r8_1d, read_attribute_char, &
           read_attribute_short_scalar, read_attribute_short_1d, &
           read_attribute_byte_scalar, read_attribute_byte_1d
   end interface read_attribute
-
+!  --- write netcdf attribute
   interface write_attribute
      module procedure write_attribute_r4_scalar, write_attribute_int_scalar,&
           write_attribute_r8_scalar, write_attribute_r4_1d,&
           write_attribute_int_1d, write_attribute_r8_1d, write_attribute_char, &
           write_attribute_short_scalar, write_attribute_short_1d, &
           write_attribute_byte_scalar, write_attribute_byte_1d
   end interface write_attribute
-
+! ---quantize variable data
   interface quantize_data
      module procedure quantize_data_2d, quantize_data_3d, &
           quantize_data_4d, quantize_data_5d
   end interface quantize_data
-
+! ---public methods
   public :: open_dataset, create_dataset, close_dataset, Dataset, Variable, Dimension, &
        read_vardata, read_attribute, write_vardata, write_attribute, get_ndim, &
        get_nvar, get_var, get_dim, get_idate_from_time_units, &
        get_time_units_from_idate, quantize_data, has_var, has_attr
-
+!
 contains
-
-  !>
+  !> check return code, print error message
   !!
   !! @param status
   !! @param halt
   !! @param fname
```

Comment:
@author tag for all subroutines/functions. Put the author tag at the end of the docblock.